smolmodels/internal/models/entities/metric.py

```diff
@@ -0,0 +1,191 @@
+"""
+Module: smolmodels/internal/common/dataclasses/metric
+
+This module defines classes for handling and comparing metrics in a flexible and extensible way.
+
+Classes:
+    - ComparisonMethod: Enum defining methods for comparing metrics.
+    - MetricComparator: Encapsulates comparison logic for metrics, including methods like higher-is-better,
+      lower-is-better, and target-is-better.
+    - Metric: Represents a specific metric with a name, value, and comparator, allowing metrics to be compared
+      and evaluated.
+
+Example Usage:
+    from metric_class import Metric, MetricComparator, ComparisonMethod
+
+    comparator = MetricComparator(ComparisonMethod.HIGHER_IS_BETTER)
+    metric1 = Metric(name="accuracy", value=0.8, comparator=comparator)
+    metric2 = Metric(name="accuracy", value=0.9, comparator=comparator)
+
+    print(metric1 < metric2)  # True
+"""
+
+from enum import Enum
+from functools import total_ordering
+
+
+class ComparisonMethod(Enum):
+    """
+    Defines methods for comparing metrics.
+
+    Attributes:
+        HIGHER_IS_BETTER: Indicates that higher values are better.
+        LOWER_IS_BETTER: Indicates that lower values are better.
+        TARGET_IS_BETTER: Indicates that values closer to a target are better.
+    """
+
+    HIGHER_IS_BETTER = "higher_is_better"
+    LOWER_IS_BETTER = "lower_is_better"
+    TARGET_IS_BETTER = "target_is_better"
+
+
+class MetricComparator:
+    """
+    Encapsulates comparison logic for metrics.
+
+    Attributes:
+        comparison_method (ComparisonMethod): The method used to compare metrics.
+        target (float, optional): The target value for TARGET_IS_BETTER comparisons.
+    """
+
+    def __init__(self, comparison_method: ComparisonMethod, target: float = None, epsilon: float = 1e-9):
+        """
+        Initializes the MetricComparator.
+
+        :param comparison_method: The method to compare metric values.
+        :param target: The target value for TARGET_IS_BETTER comparisons (optional).
+        :param epsilon: The tolerance for floating-point error in TARGET_IS_BETTER comparisons (default: 1e-9).
+        :raises ValueError: If TARGET_IS_BETTER is used without a target value.
+        """
+        self.comparison_method = comparison_method
+        self.target = target
+        self.epsilon = epsilon
+
+        if self.comparison_method == ComparisonMethod.TARGET_IS_BETTER and self.target is None:
+            raise ValueError("'TARGET_IS_BETTER' comparison requires a target value.")
+
```

Comment:
Nitpick - having a validation for `target` might make sense to ensure that it's a number 
```suggestion
if self.comparison_method == ComparisonMethod.TARGET_IS_BETTER and not isinstance(self.target, (float, int)):
    raise ValueError("'TARGET_IS_BETTER' requires a numeric target value.")

```